src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,184 @@
+from itertools import permutations
+
+import pint
+from pint import Unit
+
+base_units = pint.UnitRegistry(autoconvert_offset_to_baseunit=True)
+base_units.default_system = "SI"
+
+# Unit definitions
+base_units.define("1.e15-particles/kg = 1 kg^-1 = 1^15particles kg^-1")
+base_units.define("1.e15-particles m^-3 s^-1 = 1 m^-3 s^-1 = 1^15particles m^-3 s^-1")
+base_units.define("ton_force = 1 UK_force_ton = tonf")
+
+
+quantity = base_units.Quantity
+
+
+restricted_units = ["Hz", "hertz", "rad/s", "radian/s", "rpm", "rps", "cps"]
+restricted_conversions = list((permutations(restricted_units, 2)))
+
+
+std_prefixes = [
+    "y",
+    "z",
+    "a",
+    "f",
+    "p",
+    "n",
+    "u",
+    "c",
+    "d",
+    "da",
+    "h",
+    "m",
+    "k",
+    "M",
+    "G",
+    "T",
+    "P",
+    "E",
+    "Z",
+    "Y",
+    "",
+]
+
+
+restricted_unit_expansion = {
+    "Hz": [prefix + "Hz" for prefix in std_prefixes],
+    "hertz": [prefix + "hertz" for prefix in std_prefixes],
+    "rad/s": ["radian/s", "rad/s"],
+    "radian/s": ["rad/s", "radian/s"],
+    "rpm": ["revolutions_per_minute", "rpm"],
+    "rps": ["revolutions_per_second", "rps"],
+    "cps": ["counts_per_second", "cps"],
+}
+
+
+def build_restricted_conversions(conversions, unit_expansion):
+
+    """This function generates required final restricted mappings."""
+
+    keys = set([i[0] for i in conversions])
+    restricted_units_dict = {unit[0]: [] for unit in conversions}
+
+    for key in keys:
+        temp_list = [unit_expansion[i] for i in keys.difference(set([key]))]
+        for temp in temp_list:
+            restricted_units_dict[key] += temp
+    for key in keys:
+        restricted_units_dict[key] = list(set(restricted_units_dict[key]))
+    return restricted_units_dict
+
+
+restricted_units = build_restricted_conversions(
+    restricted_conversions, restricted_unit_expansion
+)
+
+
+class Quantity:
+
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    All the instances of this class are converted to base SI units
+    system. Any conversion between "Hz", "hertz", "rad/s", "radian/s",
```

Comment:
What does 'restricted' mean?

Reply:
The pint is able to do the conversion between above mentioned quantities but with a wrong scale factor. As we had gone through the derivation / relationship of those quantities, and we found that it's based on formula rather than a appropriate scale factor. So even if pint allows to convert them from one another produces wrong results. Hence we have restricted these conversions.

Reply:
Yes, sure.

Reply:
As well as working on the description, let's make sure that the way we are prohibiting conversions is generalizing in the required way. We are prohibiting conversions between frequency and angular velocity via control over conversions between units of those quantities. What does that mean for the treatment of radians, radians/s^2, ... ? 

As for why we don't allow the conversions, it's not that the scale factor is wrong per se; it's that we will not define a general conversion between these distinct sets of units. By raising exceptions, we rule out any ambiguities/inconsistencies. We are also consistent with the core C/C++ units library.

Reply:
Done.